### ISSUE TITLE
cannot add dotfiles to profile

### ISSUE THREAD

**Issue by ynavott** [chasinglogic/dfm#51]
I am trying to add .bashrc and my .doom.d directory to a new profile. I used `dfm init` to create a completely new profile since i don't have an existing dotfile repo. 
```
[ynavott-gram@ynavott-gram-ubuntu: ~]$ dfm init personal-dotfiles
Initialized empty Git repository in /home/ynavott-gram/.config/dfm/profiles/personal-dotfiles/.git/
[ynavott-gram@ynavott-gram-ubuntu:~]$ dfm -v link personal-dotfiles
[ynavott-gram@ynavott-gram-ubuntu:~]$ dfm add ~/.bashrc
/home/ynavott-gram/.config/dfm/profiles/personal-dotfiles:
fatal: 'origin' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```
For some reason the `link` command didn't give any output even with the `-v` option, but i tried continuing and trying to add .bashrc to the profile. it did not work. I then tried adding a directory instead of a file and got a different error.
```
[ynavott-gram@ynavott-gram-ubuntu:~]$ dfm add ~/.doom.d
Traceback (most recent call last):
  File "/usr/local/bin/dfm", line 11, in <module>
    load_entry_point('dfm==7.4.1', 'console_scripts', 'dfm')()
  File "/usr/local/lib/python3.7/dist-packages/dfm-7.4.1-py3.7.egg/dfm/cli/__init__.py", line 81, in main
    command_mod.run(docopt(command_mod.__doc__, argv=argv))
  File "/usr/local/lib/python3.7/dist-packages/dfm-7.4.1-py3.7.egg/dfm/cli/utils.py", line 71, in wrapper
    return wrapped(*args, **kwargs)
  File "/usr/local/lib/python3.7/dist-packages/dfm-7.4.1-py3.7.egg/dfm/cli/add_cmd.py", line 30, in run
    shutil.copytree(oldfile, newfile)
  File "/usr/lib/python3.7/shutil.py", line 324, in copytree
    os.makedirs(dst)
  File "/usr/lib/python3.7/os.py", line 221, in makedirs
    mkdir(name, mode)
FileExistsError: [Errno 17] File exists: '/home/ynavott-gram/.config/dfm/profiles/personal-dotfiles/..'
```
Considering the content of the first error i got for trying to add .bashrc, I tried running the add command with sudo, and the response is that there is no linked profile. 
```
[ynavott-gram@ynavott-gram-ubuntu:~]$ sudo dfm -v add ~/.bashrc
[sudo] password for ynavott-gram: 
no profile active, run dfm link to make one active
```
I then tried linking the profile using sudo and got an error as well.
```

[ynavott-gram@ynavott-gram-ubuntu:~]$ sudo dfm -v link personal-dotfiles
Traceback (most recent call last):
  File "/usr/local/bin/dfm", line 11, in <module>
    load_entry_point('dfm==7.4.1', 'console_scripts', 'dfm')()
  File "/usr/local/lib/python3.7/dist-packages/dfm-7.4.1-py3.7.egg/dfm/cli/__init__.py", line 81, in main
    command_mod.run(docopt(command_mod.__doc__, argv=argv))
  File "/usr/local/lib/python3.7/dist-packages/dfm-7.4.1-py3.7.egg/dfm/cli/link_cmd.py", line 24, in run
    profile = switch_profile(args["<profile>"])
  File "/usr/local/lib/python3.7/dist-packages/dfm-7.4.1-py3.7.egg/dfm/cli/utils.py", line 33, in switch_profile
    with open(path, "w+") as state_file:
FileNotFoundError: [Errno 2] No such file or directory: '/root/.config/dfm/state.json'
Looks like running `dfm link` with sudo isn't even supposed to work in the first place. 
```
Seems like the `link` command is to blame and doesn't execute properly and the profile isn't actually linked.

**Comment by chasinglogic** [comment]
If this is version `7.4.1` can you try `7.4.2`? Link shouldn't have any output even with `-v` if there were no files to link (as in an empty profile). I'm open to the idea that maybe it should tell you what profile it's setting as active with `-v` if you feel that it should.

It looks like you're hitting these bugs (which were fixed in `7.4.2`):

https://github.com/chasinglogic/dfm/commit/1ebf4c5d93331853b6ad880f0cd7b1e0e8af4a50
https://github.com/chasinglogic/dfm/commit/ca5cfafb658ad028ce3223aa2c6960b8cda48855

As a general rule I wouldn't sudo dfm.

**Comment by chasinglogic** [comment]
Closing due to lack of activity
